AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Migration/MutationSyncMetadataMigrationDelegate+SQLite.swift

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+import SQLite
+import AWSPluginsCore
+
+final class SQLiteMutationSyncMetadataMigrationDelegate: MutationSyncMetadataMigrationDelegate {
+
+    let modelSchemas: [ModelSchema]
+    weak var storageAdapter: SQLiteStorageEngineAdapter?
+
+    init(storageAdapter: SQLiteStorageEngineAdapter, modelSchemas: [ModelSchema]) {
+        self.storageAdapter = storageAdapter
+        self.modelSchemas = modelSchemas
+    }
+
+    func transaction(_ basicClosure: BasicThrowableClosure) throws {
+        try storageAdapter?.transaction(basicClosure)
+    }
+
+    func applyMigration(_ step: MutationSyncMetadataMigrationStep) throws {
```

Comment:
Nice! 
nit: I'd either call this `applyMigration(step:)` or `applyMigrationStep(_ step:)` just make clear we're not applying the migration multiple times

Reply:
updating with `applyMigrationStep(_ step:)`